### PR TITLE
Userposts

### PR DESCRIPTION
**Pull request by Othmanosx**

### CODE REVIEW

**Review by ZhinAbubakr — APPROVED** [review]

**Review by Ravyar19 — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
 import React from 'react'
+import Container from './containers/Container'
 
 export default function App() {
 	return (
 		<div>			
-			
-
-			
+			<Container />
```
Comment (MuhannedNoman) [inline src/App.js:7]: Indentation looks off here! Make sure to enable prettier following Airbnb style guid!

File: src/components/Dashboard/Dashboard.jsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { Avatar } from "@material-ui/core";
 import "./styles.css";
 import Post from "./Post/Post";
 
-import React, {useContext} from "react";
+import React, { useContext, useState } from "react";
```
Comment (MuhannedNoman) [inline src/components/Dashboard/Dashboard.jsx]: For convenience, any imports from inside the project come last and those from dependencies come first! This should be the first line!

File: src/components/Dashboard/Dashboard.jsx [hunk 3]
```diff
@@ -14,19 +14,24 @@ import SearchBar from "material-ui-search-bar";
...
+import { ProfileContext } from "../../profileContext";
+import { db } from "../../firebase";
+import CircularProgress from "@material-ui/core/CircularProgress";
 
 export default function Dashboard() {
-  const [profile, ] = useContext(ProfileContext)
-
+  const [profile] = useContext(ProfileContext);
+  const [posts, setPosts] = useState();
   const [state, setState] = React.useState({
```
Comment (MuhannedNoman) [inline src/components/Dashboard/Dashboard.jsx]: Why are you using `React.useState` when you have it imported already?

File: src/components/Dashboard/Dashboard.jsx [hunk 4]
```diff
@@ -14,19 +14,24 @@ import SearchBar from "material-ui-search-bar";
...
     mobileView: false,
     mediumView: false,
   });
 
   const { mobileView, mediumView } = state;
-
+  React.useEffect(() => {
```
Comment (MuhannedNoman) [inline src/components/Dashboard/Dashboard.jsx]: Again why are you using imported statements once and children again! Stick to a style, and I would say import things!

File: src/components/Dashboard/Dashboard.jsx [hunk 5]
```diff
@@ -14,19 +14,24 @@ import SearchBar from "material-ui-search-bar";
...
+    db.collection("posts").onSnapshot((snapshot) => {
+      setPosts(snapshot.docs.map((doc) => doc.data()));
+    });
+  }, []);
```
Comment (MuhannedNoman) [inline src/components/Dashboard/Dashboard.jsx]: Where is the unsubscribe method returned from Firebase to clear things up?

File: src/components/Dashboard/Dashboard.jsx [hunk 6]
```diff
@@ -14,19 +14,24 @@ import SearchBar from "material-ui-search-bar";
...
   React.useEffect(() => {
     const setResponsiveness = () => {
       if (window.innerWidth < 812) {
```
Comment (MuhannedNoman) [inline src/components/Dashboard/Dashboard.jsx]: Why is responsiveness being handled inside of a useEffect?

File: src/components/Dashboard/Dashboard.jsx [hunk 7]
```diff
@@ -85,10 +90,11 @@ export default function Dashboard() {
...
             <div className="posts">
-              <Post />
-              <Post />
-              <Post />
-              <Post />
+              {posts ? (
+                posts.map((post) => <Post post={post} />)
+              ) : (
+                <CircularProgress />
+              )}
```
Comment (MuhannedNoman) [inline src/components/Dashboard/Dashboard.jsx]: Isn't this duplicated at the end of the file? Also, why is it here?

File: src/components/Dashboard/Post/Post.jsx [hunk 8]
```diff
@@ -22,90 +21,76 @@ import ErrorIcon from "@material-ui/icons/Error";
...
+          avatar={<Avatar src={post.avatar} className={classes.avatar} />}
+          action={
+            <IconButton aria-label="settings">
+              {/* <MoreVertIcon /> */}
+              <MdMoreVert />
+            </IconButton>
+          }
+          title={post.name}
+          subheader={
+            <React.Fragment>
```
Comment (MuhannedNoman) [inline src/components/Dashboard/Post/Post.jsx]: Use <> instead of React.Fragment!

File: src/components/Dashboard/Post/Post.jsx [hunk 9]
```diff
@@ -22,90 +21,76 @@ import ErrorIcon from "@material-ui/icons/Error";
...
+          </Typography>
+          <Typography variant="body2" color="textSecondary" component="p">
+            {post.body}
+          </Typography>
+        </CardContent>
+        {/* <CardMedia
         className={classes.media}
         image="https://cosmosmagazine.com/wp-content/uploads/2020/02/190404-cat-full.jpg"
         title="Paella dish"
       /> */}
```
Comment (MuhannedNoman) [inline src/components/Dashboard/Post/Post.jsx]: Remove unused code!

**Review by MuhannedNoman — CHANGES_REQUESTED** [review]
Hey @Othmanosx , nice work here! I have left you some comments to make things better. Let me know what you think!

File: src/App.js (hunk 1 above)
Comment (Othmanosx) [reply src/App.js:7]: Prettier is working for me and code is well indented 🤷‍♂️
Screen Shot 2021-04-10 at 10 52 48 PM

File: src/components/Dashboard/Dashboard.jsx (hunk 2 above)
Comment (Othmanosx) [reply src/components/Dashboard/Dashboard.jsx]: Done 140ca9f9a8e00f3dd1f9e0803e6fff7bbca30740

File: src/components/Dashboard/Dashboard.jsx (hunk 3 above)
Comment (Othmanosx) [reply src/components/Dashboard/Dashboard.jsx]: Done 783ee049a3396f564058a194d966ebf9d3ca9f7e

File: src/components/Dashboard/Dashboard.jsx (hunk 4 above)
Comment (Othmanosx) [reply src/components/Dashboard/Dashboard.jsx]: Fixed 783ee049a3396f564058a194d966ebf9d3ca9f7e

File: src/components/Dashboard/Dashboard.jsx (hunk 5 above)
Comment (Othmanosx) [reply src/components/Dashboard/Dashboard.jsx]: Fixed 140ca9f9a8e00f3dd1f9e0803e6fff7bbca30740

File: src/components/Dashboard/Dashboard.jsx (hunk 6 above)
Comment (Othmanosx) [reply src/components/Dashboard/Dashboard.jsx]: Fixed a84d1cf67350943217980232d65e037e00efcb9e

File: src/components/Dashboard/Dashboard.jsx (hunk 7 above)
Comment (Othmanosx) [reply src/components/Dashboard/Dashboard.jsx]: Fixed 783ee049a3396f564058a194d966ebf9d3ca9f7e

File: src/components/Dashboard/Post/Post.jsx (hunk 8 above)
Comment (Othmanosx) [reply src/components/Dashboard/Post/Post.jsx]: Fixed 7c3841e95070633f0f9b2f8da4bb63e578653590

File: src/components/Dashboard/Post/Post.jsx (hunk 9 above)
Comment (Othmanosx) [reply src/components/Dashboard/Post/Post.jsx]: Fixed 7c3841e95070633f0f9b2f8da4bb63e578653590

**Pull request merged**